### PR TITLE
Support an explicit rollback action

### PR DESCRIPTION
**Pull request by ajordens**
- `Rollback` becomes an server group action on enabled server groups

### CODE REVIEW

**Comment by ajordens** [comment]
PTAL @anotherchrisberry.

As far as those directives go, I pulled them up a bit but left them in `aws`. I'm sure there's a more appropriate long-term place.

**Comment by ajordens** [comment]
image

image

File: app/scripts/modules/amazon/common/footer.directive.js [hunk 1]
```diff
@@ -3,15 +3,15 @@
 let angular = require('angular');
 
 module.exports = angular
-  .module('spinnaker.amazon.serverGroup.details.resize.footer.directive', [
+  .module('spinnaker.amazon.footer.directive', [
   ])
-  .directive('awsResizeFooter', function () {
+  .directive('awsFooter', function () {
```
Comment (anotherchrisberry) [inline app/scripts/modules/amazon/common/footer.directive.js]: This should probably be "modalFooter", if we are going to abstract it away from the resize modal, but I will do that as part of a separate PR this afternoon.

File: app/scripts/modules/amazon/serverGroup/details/rollback/rollbackServerGroup.html [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+          </ui-select>
+        </div>
+      </div>
+
+      <div class="row" ng-if="command.platformHealthOnlyShowOverride">
+        <div class="col-sm-10 col-sm-offset-1">
+          <platform-health-override command="command"
+                                    platform-health-type="'Amazon'"
+                                    show-help-details="true"
+                                    field-columns="12">
```
Comment (anotherchrisberry) [inline app/scripts/modules/amazon/serverGroup/details/rollback/rollbackServerGroup.html]: `field-columns` doesn't do anything any more (but I need to update the directive definition, so I will clean this up at that point).

File: app/scripts/modules/core/serverGroup/serverGroup.write.service.js [hunk 3]
```diff
@@ -60,6 +60,23 @@ module.exports = angular
       });
     }
 
+    function rollbackServerGroup(serverGroup, application, params={}) {
+      params.asgName = serverGroup.name;
+      params.serverGroupName = serverGroup.name;
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/serverGroup/serverGroup.write.service.js]: Do we need both here? You just wrote the Orca functionality - does it expect the "asgName" field to be populated from the client?

File: app/scripts/modules/core/serverGroup/serverGroup.write.service.js [hunk 4]
```diff
@@ -60,6 +60,23 @@ module.exports = angular
...
+      params.type = 'rollbackServerGroup';
+      params.regions = [serverGroup.region];
+      params.region = serverGroup.region;
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/serverGroup/serverGroup.write.service.js]: Same as above - do we need `regions` and `region`?

Comment (ajordens) [inline app/scripts/modules/core/serverGroup/serverGroup.write.service.js]: `orca` is just passing almost directly through to cloud driver hence following the same convention here.

Just looking at CD to see if we can't tidy up some of this.

**Pull request merged**